core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1276,6 +1276,10 @@ private SqlNode toSql(SqlOperator op, boolean distinct,
         return preOperandList.get(0);
       }
 
+      if (op == SqlInternalOperators.CAST_NOT_NULL) {
+        return preOperandList.get(0);
+      }
+
```

Comment:
My understanding is that the new operator should never make it up to this stage. Am I missing something?